packages/atomic/src/components/insight/atomic-insight-result/atomic-insight-result.tsx

```diff
@@ -123,6 +123,34 @@ export class AtomicInsightResult {
       .join('');
   }
 
+  private handleMouseOver = () => {
+    this.resultActionBar?.classList.add('showActionBar');
+    this.host.classList.add('resultHovered');
+  };
+
+  private handleMouseOut = () => {
+    this.resultActionBar?.classList.remove('showActionBar');
+    this.host.classList.remove('resultHovered');
+  };
+
+  private get resultActionBar() {
+    return this.host.shadowRoot?.querySelector(
+      'atomic-insight-result-action-bar'
+    );
+  }
+
+  private get resultActions() {
+    return this.host.shadowRoot?.querySelectorAll(
+      'atomic-insight-result-action'
+    );
+  }
+
+  private get firstChildActionBar() {
+    return this.host.parentElement
+      ?.querySelectorAll('atomic-insight-result')[0]
```

Comment:
are we missing an optional chainer here? for when the array is empty

Reply:
the getter is called after checking that we have `this.resultActions?.length` but I could add the optional chainer also here :)

Reply:
Yeah IMO a getter should not be implemented based on where it's called. Once it's a getter is should be a secure and reliable way to get a value on its own. If we add code in the future that uses the getter as well it would be unnecessarily difficult to remember this requirement to check for `this.resultActions?.length` before using it.